crd/multitenantpodnetworkconfig/api/v1alpha1/multitenantpodnetworkconfig.go

```diff
@@ -0,0 +1,62 @@
+//go:build !ignore_uncovered
+// +build !ignore_uncovered
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// Important: Run "make" to regenerate code after modifying this file
+
+// +kubebuilder:object:root=true
+
+// MultitenantPodNetworkConfig is the Schema for the multitenantpodnetworkconfigs API
+// +kubebuilder:resource:scope=Namespaced
+// +kubebuilder:resource:shortName=mpnc
+// +kubebuilder:subresource:status
+// +kubebuilder:metadata:labels=managed=
+// +kubebuilder:metadata:labels=owner=
+// +kubebuilder:printcolumn:name="PodNetworkInstance",type=string,JSONPath=`.spec.podNetworkInstance`
+// +kubebuilder:printcolumn:name="PodName",type=string,JSONPath=`.spec.podName`
+// +kubebuilder:printcolumn:name="UUID",type=string,JSONPath=`.status.uuid`
+// +kubebuilder:printcolumn:name="PrimaryIP",type=string,JSONPath=`.status.primaryIP`
+// +kubebuilder:printcolumn:name="MacAddress",type=string,JSONPath=`.status.macAddress`
+type MultitenantPodNetworkConfig struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   MultitenantPodNetworkConfigSpec   `json:"spec,omitempty"`
+	Status MultitenantPodNetworkConfigStatus `json:"status,omitempty"`
+}
+
+// +kubebuilder:object:root=true
+
+// MultitenantPodNetworkConfigList contains a list of PodNetworkConfig
+type MultitenantPodNetworkConfigList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []MultitenantPodNetworkConfig `json:"items"`
+}
+
+// MultitenantPodNetworkConfigSpec defines the desired state of PodNetworkConfig
+type MultitenantPodNetworkConfigSpec struct {
+	// name of PNI object from requesting cx pod
+	PodNetworkInstance string `json:"podNetworkInstance,omitempty"`
+	// name of the requesting cx pod
+	PodName string `json:"podName,omitempty"`
+}
+
+// MultitenantPodNetworkConfigStatus defines the observed state of PodNetworkConfig
+type MultitenantPodNetworkConfigStatus struct {
+	// network container id
+	UUID string `json:"uuid,omitempty"`
+	// ip allocated to the network container
+	PrimaryIP string `json:"primaryIP,omitempty"`
```

Comment:
You also need to pass the IPSubnet details

Reply:
Waiting for Jaeryn to update the spec if this is needed

Reply:
For now let's just add `Gateway string `json:"gateway,omitempty"`. Will update the spec after we close the routing design discussion, and if we need more than gateway we can add it in a future PR. Don't want to block RC dev until closing routing design and that design will impact this. So let's just add this one property and close this.